src/Storages/MergeTree/MergeTreeIndexFullText.cpp

```diff
@@ -548,12 +548,21 @@ bool MergeTreeConditionFullText::traverseTreeEquals(
         token_extractor->stringLikeToBloomFilter(value.data(), value.size(), *out.bloom_filter);
         return true;
     }
-    else if (function_name == "hasToken")
+    else if (function_name.starts_with("hasToken"))
     {
         out.key_column = key_column_num;
         out.function = RPNElement::FUNCTION_EQUALS;
         out.bloom_filter = std::make_unique<BloomFilter>(params);
-        const auto & value = const_value.get<String>();
+        const auto value = std::invoke(
+            [&const_value, &function_name]
+            {
+                auto str = const_value.get<String>();
+                if (function_name.starts_with("hasTokenCaseInsensitive"))
+                    std::ranges::transform(str, str.begin(), &toLowerIfAlphaASCII);
```

Comment:
Looking at the implementation of `toLowerIfAlphaASCII`, this will work fine for characters "a-zA-Z" but blow up for "0-9". E.g. 5 will convert to U. The standard tokenizer in ClickHouse will split across non-alphanumeric characters, so tokens like `ab23CD` are possible.

Reply:
Indeed, thanks for noticing!